### PR TITLE
static config options for basic litecoin support

### PR DESCRIPTION
**Pull request by Saibato**
to make LTC fully work more changes are probably necessary ;-)

### CODE REVIEW

**Comment by Saibato** [comment]
yes  @ZmnSCPxj  good point litecoin has 4 times more blockcount/hour so I changed this wait values too.
I guess there must be even more adaption maybe we should ask @NicolasDorier what he needs for LTC in his probably skyrocketing btcpay Lapp.

File: lightningd/options.c [hunk 1]
```diff
@@ -493,6 +493,125 @@ static const struct config mainnet_config = {
...
+	 * Each node SHOULD flush outgoing announcements once every 60 seconds */
+	.broadcast_interval = 60000,
+
+	/* Send a keepalive update at least every week, prune every twice that */
+	.channel_update_interval = 1209600/2,
+
+	/* Mainnet should have more stable fees */
+	.ignore_fee_limits = false,
+
+	/* Rescan 2.5 hours of blocks on startup, it's not so reorg happy */
```
[the remark this answers is not in the document]
Comment (Saibato) [reply lightningd/options.c]: @ZmnSCPxj changes done

File: lightningd/options.c [hunk 2]
```diff
@@ -493,6 +493,125 @@ static const struct config mainnet_config = {
...
+	 * Each node SHOULD flush outgoing announcements once every 60 seconds */
+	.broadcast_interval = 60000,
+
+	/* Send a keepalive update at least every week, prune every twice that */
+	.channel_update_interval = 1209600/2,
+
+	/* Testnet sucks */
+	.ignore_fee_limits = true,
+
+	/* Rescan 5 hours of blocks on testnet, it's reorg happy */
```
[the remark this answers is not in the document]
Comment (Saibato) [reply lightningd/options.c]: @ZmnSCPxj changes done

[the remark this answers is not in the document]
Comment (ZmnSCPxj) [reply lightningd/options.c]: Hmm was it pushed to github?

**Comment by Saibato** [comment]
@cdecker just to your info forced pushed this simple change on top of master 3 times with no code changes 2 times CI failed. :question:

**Comment by NicolasDorier** [comment]
cool! finally I can test litecoin on regtest! :)

I did not tried LTC on LN with BTCPay, (even if it is supported in the docker install) so I don't really know what is needed to make it works better between implentations.

**Comment by Saibato** [comment]
@NicolasDorier please warn everybody who tests this that this is not fully supported and no doku at all.
And there is no LTC display so users can get confused and feel richer then they are :smile_cat: 

regtest worked but i must gen a lot of blocks for turning steps, maybe you can figure out and improve?

File: lightningd/options.c [hunk 3]
```diff
@@ -592,7 +592,7 @@ static const struct config litecoin_testnet_config = {
 	.ignore_fee_limits = true,
 
 	/* Rescan 5 hours of blocks on testnet, it's reorg happy */
-	.rescan = 30,
+	.rescan = 120,
```
Comment (cdecker) [inline lightningd/options.c]: Needs to be squashed

File: bitcoin/chainparams.c [hunk 4]
```diff
@@ -53,6 +53,16 @@ const struct chainparams networks[] = {
      .cli_args = "-testnet",
      .dust_limit = 100000,
      .when_lightning_became_cool = 1,
+     .testnet = true},
+      {.index = 5,
```
Comment (cdecker) [inline bitcoin/chainparams.c]: NACK, don't need to add even more networks, I'd remove litecoin-testnet as well, testing should always be done on bitcoin testnet and regtest.

File: bitcoin/chainparams.c [hunk 5]
```diff
@@ -56,7 +56,7 @@ const struct chainparams networks[] = {
      .testnet = true},
       {.index = 5,
      .network_name = "litecoin-regtest",
-     .bip173_name = "lcrt",
+     .bip173_name = "rltc",
```
Comment (cdecker) [inline bitcoin/chainparams.c]: Needs to be squashed

File: bitcoin/chainparams.c (hunk 4 above)
Comment (Saibato) [reply bitcoin/chainparams.c]: regtest for LTC is for development and easy test essential
in litecoin rltc is the bech32 tag

File: bitcoin/chainparams.c (hunk 5 above)
Comment (Saibato) [reply bitcoin/chainparams.c]: done

File: bitcoin/chainparams.c (hunk 4 above)
Comment (NicolasDorier) [reply bitcoin/chainparams.c]: I think regtest for litecoin should be supported:

1. it does not cost too much to add.
2. If I want to build something based on clightning for making atomic swap on lightning, I need at least 2 regtest.

**Pull request closed without merging**